commit ef0d953d9e2d
Author: Makoto Kato <m_kato@ga2.so-net.ne.jp>
Date:   Tue Aug 23 16:54:06 2016 +0900

    Bug 1142056 - Add aarch64 configuration on Skia's moz.build. r=gw280
    
    MozReview-Commit-ID: 60iAZjFg6W6
    
    --HG--
    extra : histedit_source : b3bb5a3e556393ba49f43dc4ab8bdaa9be40fb0c
---
 gfx/skia/generate_mozbuild.py |  6 ++++--
 gfx/skia/moz.build            | 12 ++++++++++--
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git gfx/skia/generate_mozbuild.py gfx/skia/generate_mozbuild.py
index 2088b729d476..1eb04c1d7f57 100755
--- gfx/skia/generate_mozbuild.py
+++ gfx/skia/generate_mozbuild.py
@@ -429,10 +429,12 @@ def write_mozbuild(sources):
   f.write("if CONFIG['INTEL_ARCHITECTURE']:\n")
   write_sources(f, sources['intel'], 4)
 
-  f.write("elif CONFIG['CPU_ARCH'] == 'arm' and CONFIG['GNU_CC']:\n")
+  f.write("elif CONFIG['CPU_ARCH'] in ('arm', 'aarch64') and CONFIG['GNU_CC']:\n")
   write_sources(f, sources['arm'], 4)
 
-  f.write("    if CONFIG['BUILD_ARM_NEON']:\n")
+  f.write("    if CONFIG['CPU_ARCH'] == 'aarch64':\n")
+  write_sources(f, sources['neon'], 8)
+  f.write("    elif CONFIG['BUILD_ARM_NEON']:\n")
   write_list(f, 'SOURCES', sources['neon'], 8)
   write_cflags(f, sources['neon'], 'neon', "CONFIG['NEON_FLAGS']", 8)
 
diff --git gfx/skia/moz.build gfx/skia/moz.build
index b33f1cdbee54..9dba87978dd3 100644
--- gfx/skia/moz.build
+++ gfx/skia/moz.build
@@ -519,7 +519,7 @@ if CONFIG['INTEL_ARCHITECTURE']:
         'skia/src/opts/SkOpts_sse41.cpp',
         'skia/src/opts/SkOpts_ssse3.cpp',
     ]
-elif CONFIG['CPU_ARCH'] == 'arm' and CONFIG['GNU_CC']:
+elif CONFIG['CPU_ARCH'] in ('arm', 'aarch64') and CONFIG['GNU_CC']:
     UNIFIED_SOURCES += [
         'skia/src/core/SkUtilsArm.cpp',
         'skia/src/opts/SkBitmapProcState_opts_arm.cpp',
@@ -528,7 +528,15 @@ elif CONFIG['CPU_ARCH'] == 'arm' and CONFIG['GNU_CC']:
     SOURCES += [
         'skia/src/opts/SkBlitRow_opts_arm.cpp',
     ]
-    if CONFIG['BUILD_ARM_NEON']:
+    if CONFIG['CPU_ARCH'] == 'aarch64':
+        SOURCES += [
+            'skia/src/opts/SkBitmapProcState_arm_neon.cpp',
+            'skia/src/opts/SkBitmapProcState_matrixProcs_neon.cpp',
+            'skia/src/opts/SkBlitMask_opts_arm_neon.cpp',
+            'skia/src/opts/SkBlitRow_opts_arm_neon.cpp',
+            'skia/src/opts/SkOpts_neon.cpp',
+        ]
+    elif CONFIG['BUILD_ARM_NEON']:
         SOURCES += [
             'skia/src/opts/SkBitmapProcState_arm_neon.cpp',
             'skia/src/opts/SkBitmapProcState_matrixProcs_neon.cpp',
